features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const request = require('../support/request');
+const { defineSupportCode } = require('cucumber');
+
+/**
+ * Helper function to create a collection
+ * @method createCollection
+ * @param   {Object}  body    The body of the request
+ * @returns {Promise}
+ */
+function createCollection(body) {
+    return this.getJwt(this.apiToken)
+        .then((response) => {
+            this.jwt = response.body.token;
+
+            return request({
+                uri: `${this.instance}/${this.namespace}/collections`,
+                method: 'POST',
+                auth: {
+                    bearer: this.jwt
+                },
+                body,
+                json: true
+            });
+        });
+}
+
+/**
+ * Helper function to delete a collection
+ * @method deleteCollection
+ * @param   {Number} id  Id of the collection to delete
+ * @returns {Promise}
+ */
+function deleteCollection(id) {
+    if (!id) {
+        return Promise.resolve();
+    }
+
+    return request({
+        uri: `${this.instance}/${this.namespace}/collections/${id}`,
+        method: 'DELETE',
+        auth: {
+            bearer: this.jwt
+        },
+        json: true
+    }).then((response) => {
+        Assert.oneOf(response.statusCode, [204, 404]);
+    });
+}
+
+defineSupportCode(({ Before, Given, Then, When, After }) => {
+    Before('@collections', function hook() {
+        this.repoOrg = this.testOrg;
+        this.repoName = 'functional-collections';
+        this.pipelineId = null;
+        this.firstCollectionId = null;
+        this.secondCollectionId = null;
+    });
+
+    When(/^they create a new collection "myCollection" with that pipeline$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName })
+            .then(() => {
+                const requestBody = {
+                    name: 'myCollection',
+                    pipelineIds: [this.pipelineId]
+                };
+
+                return createCollection.call(this, requestBody)
+                    .then((response) => {
+                        Assert.oneOf(response.statusCode, [409, 201]);
```

Comment:
This is supposed to assert that the collection "myCollection" is newly created. Why is `409` an acceptable response code?

Reply:
Sorry I was referencing some other tests when writing this and it had a check for `409` but I see now why that was allowed in those tests. I'll make the change